test-result-summary-client/src/PerfCompare/lib/JenkinsRunJSON.js

```diff
@@ -26,57 +23,23 @@ export default class JenkinsRunJSON {
                 // new variant
                 if (parsedVariantsCommon[curVariantObjectId] === undefined) {
                     parsedVariantsCommon[curVariantObjectId] = {
-                        product: (this.runJSON.testInfo[0].tests[k].benchmarkProduct === undefined) ? null : this.runJSON.testInfo[0].tests[k].benchmarkProduct,
-                        result: (this.runJSON.testInfo[0].tests[k].testResult === undefined) ? null : this.runJSON.testInfo[0].tests[k].testResult,
-                        benchmark: (this.runJSON.testInfo[0].tests[k].benchmarkName === undefined) ? null : this.runJSON.testInfo[0].tests[k].benchmarkName,
-                        variant: (this.runJSON.testInfo[0].tests[k].benchmarkVariant === undefined) ? null : this.runJSON.testInfo[0].tests[k].benchmarkVariant,
-                        machine: (this.runJSON.testInfo[0].machine === undefined) ? null : this.runJSON.testInfo[0].machine,
-                        metrics: (this.runJSON.testInfo[0].tests[k].testData.metrics === undefined) ? [] : this.runJSON.testInfo[0].tests[k].testData.metrics
+                        product: (aggregateInfo.benchmarkProduct === undefined) ? null : aggregateInfo.benchmarkProduct,
+                        result: (aggregateInfo.testResult === undefined) ? null : aggregateInfo.testResult,
+                        benchmark: (aggregateInfo.benchmarkName === undefined) ? null : aggregateInfo.benchmarkName,
+                        variant: (aggregateInfo.benchmarkVariant === undefined) ? null : aggregateInfo.benchmarkVariant,
+                        machine: (this.runJSON.testInfo[0].machine === null) ? null : this.runJSON.testInfo[0].machine.split('.')[0],
+                        metrics: (aggregateInfo.metrics === undefined) ? [] : aggregateInfo.metrics,
+                        testsData: (this.runJSON.testInfo[0].tests === undefined) ? undefined : this.runJSON.testInfo[0].tests[0].testData.metrics
```

Comment:
Need to take care of the scenario when we're running more than one test in one Jenkins job.